packages/block-editor/src/components/global-styles/utils.js

```diff
@@ -415,6 +415,7 @@ export function areGlobalStyleConfigsEqual( original, variation ) {
 	}
 	return (
 		fastDeepEqual( original?.styles, variation?.styles ) &&
-		fastDeepEqual( original?.settings, variation?.settings )
+		fastDeepEqual( original?.settings, variation?.settings ) &&
+		fastDeepEqual( original?.behaviors, variation?.behaviors )
```

Comment:
This is making [the e2e test](https://github.com/WordPress/gutenberg/blob/ebe60b44c0323ee052ee716007dd85cc97d2a650/test/e2e/specs/site-editor/style-variations.spec.js#L55-L58) fail because `original.behaviors === {}` and `variation.behaviors === undefined`. 

Are we sure that this is needed? If yes, then the fixtures for the test probably need to be updated.

Reply:
We need it, otherwise revisions with only behaviors won't work.

Reply:
gotcha. I found the issue, we needed to add a behaviors property here: [ea1d540](https://github.com/WordPress/gutenberg/pull/52370/commits/ea1d54053ce4941a6049cef437c4047ba17fbaa7)